### ISSUE TITLE
Invalid code when generating both POJOs and interfaces with UDTs

### ISSUE THREAD

**Issue by rmdefi** [jOOQ/jOOQ#8209]
I tried generating both POJOs and interfaces for a schema containing a custom composite type.

Here is the schema (for PostgreSQL) used to test:

```sql
CREATE TYPE public.compo AS (
        c1 integer,
        c2 text
);
CREATE TABLE public.tbl (
    id integer NOT NULL,
    val public.compo
);
```

Here is the configuration I used in gradle for the jooq plugin:

```gradle
	generator {
		generate {
			daos                     = false
			interfaces               = true
			globalObjectReferences   = false
			pojos                    = true
			records                  = true
			generatedAnnotation      = false
			javaTimeTypes            = true
		}
		target {
			directory='src/db/java'
			packageName='db'
		}
		database {
			name='org.jooq.meta.postgres.PostgresDatabase'
			inputSchema='public'
		}
	}
```
See attached file with the full generated code
[db.zip](https://github.com/jOOQ/jOOQ/files/2750117/db.zip)

The problem is in the class db.tables.pojos.Tbl, for the constructor taking the corresponding interface as parameter:

```java
    public Tbl(ITbl value) {
        this.id = value.getId();
        this.val = value.getVal();
    }
```
`this.val` is of type Compo (the POJO of my composite type), but `value.getVal()` returns an ICompo (the interface of that same type).

A quick way to fix this is to create a new instance of Comp, but then we must check for null values, example:
```java
this.val = value.getVal()==null ? null : new Compo(value.getVal());
```

### Versions:

- jOOQ: 3.11.9
- Java: 1.8
- Database (include vendor): PostgreSQL 9.6
- gradle: 4.10.3
- gradle plugin nu.studer.jooq: 3.0.2

**Comment by lukaseder** [comment]
Thank you very much for your report. I'm sorry for the delay in the response. The notification email seems to have gone lost. I think this is a duplicate of a known issue but I cannot seem to find the other one.

No further steps to reproduce this are needed. Unfortunately, I can't say when this can be fixed, there are many edge cases in this area, when generating interfaces.

**Comment by axelfontaine** [comment]
Note that the equivalent `TblRecord.from(ITbl from)` method also fails to compile for the same reason.

**Comment by lukaseder** [comment]
I've been fixing related issues when using interfaces on embeddable records:

- https://github.com/jOOQ/jOOQ/issues/11716
- https://github.com/jOOQ/jOOQ/issues/17165
- https://github.com/jOOQ/jOOQ/issues/17180

Embeddables are "UDT emulations" and work in a very similar way, so perhaps, this can be solved more easily now.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.20.0

There have been significant changes in this area due to the aforementioned issues, which makes a backport unrealistic.
